src/diffusers/configuration_utils.py

```diff
@@ -706,3 +706,32 @@ def init(self, *args, **kwargs):
 
     cls.__init__ = init
     return cls
+
+
+class LegacyConfigMixin(ConfigMixin):
+    r"""
+    A subclass of `ConfigMixin` to resolve class mapping from legacy classes (like `Transformer2DModel`) to more
+    pipeline-specific classes (like `DiTTransformer2DModel`).
+    """
+
+    @classmethod
+    def from_config(cls, config: Union[FrozenDict, Dict[str, Any]] = None, return_unused_kwargs=False, **kwargs):
+        # resolve remapping
+        if cls.__name__ == "Transformer2DModel":
+            # prevent circular imports
+            from .models.transformers import DiTTransformer2DModel, PixArtTransformer2DModel
```

Comment:
I think we employ a similar mapping strategy using a dict as we did with the pipelines. And I think importing the object using something similar to this:
https://github.com/huggingface/diffusers/blob/e1df77ee1ec0400195ad8dcb2099d137b34c9b9f/src/diffusers/loaders/single_file.py#L69